### PR TITLE
[FEATURE] Support both UserConfigurableProfiler and OnboardingDataAssistant in "CLI SUITE NEW --PROFILE <name>" command

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* For a smoother rollout, both, `UserConfigurableProfiler` and `OnboardingDataAssistant` in "CLI SUITE NEW --PROFILE <name>" command are now supported.  When <name> is absent, the `UserConfigurableProfiler` is launched in the `Jupyter` notebook..  When <name> is "onboarding-data-assistant", the `OnboardingDataAssistant` in the `Jupyter` notebook.  When `UserConfigurableProfiler` is finally replaced by `OnboardingDataAssistant`, the latter will be invoked using the same way as is currently the former (i.e., CLI SUITE NEW --PROFILE with the <name> is absent).  When <name> is present, the named `Rule-Based Profiler` is launched (this behavior has stayed unaltered and will stay the same).
* Updated unit tests.
* Added "TODO" comments to attend to for the final update from `UserConfigurableProfiler` to `OnboardingDataAssistant`.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-464/GREAT-994
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/render/renderer/suite_scaffold_notebook_renderer.py [hunk 1]
```diff
@@ -169,7 +169,7 @@ def _add_scaffold_cell(self) -> None:
     ignored_columns=ignored_columns,
     excluded_expectations=None,
     not_null_only=False,
-    primary_or_compound_key=False,
+    primary_or_compound_key=None,
```
Comment (anthonyburdi) [inline great_expectations/render/renderer/suite_scaffold_notebook_renderer.py:172]: Good catch

File: tests/cli/test_suite.py [hunk 2]
```diff
@@ -24,12 +24,76 @@
...
+    # include_column_names=include_column_names,
+    exclude_column_names=exclude_column_names,
+    # include_column_name_suffixes=include_column_name_suffixes,
+    # exclude_column_name_suffixes=exclude_column_name_suffixes,
+    # semantic_type_filter_module_name=semantic_type_filter_module_name,
+    # semantic_type_filter_class_name=semantic_type_filter_class_name,
+    # include_semantic_types=include_semantic_types,
+    # exclude_semantic_types=exclude_semantic_types,
+    # allowed_semantic_types_passthrough=allowed_semantic_types_passthrough,
+    cardinality_limit_mode="rel_100",  # case-insenstive (see documentaiton for other options)
```
Comment (anthonyburdi) [inline tests/cli/test_suite.py:62]: Spelling

File: great_expectations/render/renderer/v3/suite_profile_notebook_renderer.py [hunk 3]
```diff
@@ -127,6 +131,112 @@ def _add_user_configurable_profiler_cells(self) -> None:
...
+""",
+            lint=True,
+        )
+        self.add_markdown_cell(
+            markdown="""\
+# Select columns
+
+Select the columns on which you would like to set expectations and those which you would like to ignore.
+
+Great Expectations will choose which expectations might make sense for a column based on the **data type** and **cardinality** of the data in each selected column.
```
Comment (NathanFarmer) [inline great_expectations/render/renderer/v3/suite_profile_notebook_renderer.py:165]: Is a comprehensive description for this profiler?

File: great_expectations/render/renderer/suite_scaffold_notebook_renderer.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/render/renderer/suite_scaffold_notebook_renderer.py:172]: @anthonyburdi with a little help from `PyCharm` :)

File: great_expectations/render/renderer/v3/suite_profile_notebook_renderer.py (hunk 3 above)
Comment (NathanFarmer) [reply great_expectations/render/renderer/v3/suite_profile_notebook_renderer.py:165]: I see the same description for `OnboardingDataAssistant` in `_add_onboarding_data_assistant_cells`. I think this sentence either needs to be more precise, or more general, for both profilers.

**Review by anthonyburdi — APPROVED** [review]
LGTM!

File: tests/cli/test_suite.py (hunk 2 above)
Comment (NathanFarmer) [inline tests/cli/test_suite.py:62]: Something smells wrong about how this is the only config not commented out.

**Review by NathanFarmer — COMMENTED** [review]
I have a few requests for another PR!

File: tests/cli/test_suite.py (hunk 2 above)
Comment (alexsherstinsky) [reply tests/cli/test_suite.py:62]: @NathanFarmer Fixing in #5310 -- thanks!

**Pull request merged**